samples/basic/minimal/sample.yaml

```diff
@@ -9,81 +9,86 @@ common:
     regex:
       - "Hello World from minimal!"
 tests:
-  sample.minimal.mt.arm:
-    extra_args: CONF_FILE='common.conf;mt.conf;arm.conf'
+  sample.minimal.mt:
+    arch_allow:
+      - arm
```

Comment:
why `arch_allow` is good but not `plaform_allow`?

Reply:
> why `arch_allow` is good but not `platform_allow`?

This is better than adding a list of all arm and x86 boards.

Reply:
I find arc and riscv has its own minimal test config already. so I assumption is that the minimal setting config is arch depended.
```
  sample.minimal.arc.runtime:
    extra_args:
      - CONF_FILE='common-runtime.conf;no-timers.conf'
    arch_allow: arc
    tags:
      - kernel
    integration_platforms:
      - qemu_arc/qemu_arc_em
  sample.minimal.riscv.runtime:
    extra_args:
      - CONF_FILE='common-runtime.conf;no-timers.conf;no-mt.conf;riscv.conf'
    platform_allow: qemu_riscv32
    tags:
      - kernel
    integration_platforms:
      - qemu_riscv32
```